turbopack/crates/turbopack-ecmascript/tests/analyzer/graph/array/resolved-explained.snapshot

```diff
@@ -31,3 +31,11 @@ d3 = ("foo" | ???*0*)
 d4 = ???*0*
 - *0* [1, "foo"][2]
   ⚠️  invalid index
+
+value#3 = (1 | "foo" | ???*0*)
+- *0* unknown mutation
+  ⚠️  This value might have side effects
+
+value#5 = ???*0*
+- *0* unknown mutation
+  ⚠️  This value might have side effects
```

Comment:
This could in fact be improved (for the literal `for (const value of [])` case)
So a array literal could somehow be `mutable: false` but then variable assignments and function calls would make it mutable.
Which is in effect escape analysis.